core/server/api/subscribers.js

```diff
@@ -281,83 +279,42 @@ subscribers = {
+                            duplicates = duplicates + 1;
                         } else {
-                            emailIdx = _.findIndex(dataToImport, function (columnName) {
-                                if (columnName.match(/email/g)) {
-                                    return true;
-                                } else {
-                                    return false;
-                                }
-                            });
-                        }
-
-                        if (emailIdx === -1) {
-                            return reject(new errors.ValidationError(
-                                'Couldn\'t find your email addresses! Please use a column header which contains the word "email".'
-                            ));
+                            invalid = invalid + 1;
                         }
-                        firstLine = false;
-                    } else if (emailIdx > -1) {
-                        importTasks.push(function () {
-                            return subscribers.add({
-                                subscribers: [{
-                                    email: dataToImport[emailIdx]
-                                }
-                            ]}, {context: options.context});
-                        });
                     }
                 });
-
-                rl.on('close', function () {
-                    var fulfilled = 0,
-                        duplicates = 0,
-                        invalid = 0;
-
-                    Promise.all(importTasks.map(function (promise) {
-                        return promise().reflect();
-                    })).each(function (inspection) {
-                        if (inspection.isFulfilled()) {
-                            fulfilled = fulfilled + 1;
-                        } else {
-                            if (inspection.reason().errorType === 'ValidationError') {
-                                invalid = invalid + 1;
-                            } else if (inspection.reason().errorType === 'DataImportError') {
-                                duplicates = duplicates + 1;
-                            }
-                        }
-                    }).then(function () {
-                        return resolve({
-                            stats: [{
-                                imported: fulfilled,
-                                duplicates: duplicates,
-                                invalid: invalid
-                            }]
-                        });
-                    }).catch(function (err) {
-                        return reject(err);
-                    }).finally(function () {
-                        // Remove uploaded file from tmp location
-                        return pUnlink(filePath);
-                    });
-                });
+            }).then(function () {
+                return {
+                    stats: {
```

Comment:
This returned an array previously in order to be consistent with all other API endpoints which return data in the format:

`{key: [{...}, ...]}`

Buuuuut the key is meant to match the endpoint, so the `subscribers` endpoint should really have a key of `subscribers`. However, in this case, we're not returning subscribers, but some meta information. This makes me think we should be returning: `meta: {stats: {}}` rather than stats at the top level? 

According to http://jsonapi.org/ (which we very loosely follow) we are only meant to have a `data`, `errors` or `meta` key and we're currently stuck on a previous iteration where the `data` key was named for the resource/endpoint.

cc @sebgie who is usually in charge of API design 😉